### DISCUSSION TITLE
Performance of the Netflix Conductor over the period of time

### DISCUSSION THREAD

**Discussion by npk1994** [Netflix/conductor#2932] [Q&A] [answered]
Hello Everyone,

We have Cloud Application with Microservice architecture. We have around 30-45 microservices(with multiple apis ) to start with.  We are on track to implement conductor for orchestration. We are planning to implement Oracle DB for persistence. Over the period of time of as we bring in more workflows into the system, the data set will increases. We would like to know how to ensure data pertaining to older executions are not impacting in a long run of tasks.

We would also like to understand if there are any mechanism already in place to tackle the performance issues.

Appreciate your suggestions. 
Thank you!

**Comment by c4lm** [comment, marked as answer]
Everything depends on the number and the size of the workflows you're running. If you run a lot of workflows, you should consider using table partitioning (and maybe caching - for example, to display last N workflow runs in the UI); if the workflows are large in size, consider archiving old workflows in object storage (keep in mind that clouds typically charge for all I/O operations, so you might want to store batches of workflows to reduce I/O count)
